csharp/src/Microsoft.ML.OnnxRuntime/NativeMethods.cs

```diff
@@ -560,6 +600,16 @@ static NativeMethods()
         [DllImport(nativeLib, CharSet = charSet)]
         public static extern IntPtr /*(OrtStatus*)*/ OrtSessionOptionsAppendExecutionProvider_CUDA(IntPtr /*(OrtSessionOptions*) */ options, int device_id);
 
+        /// <summary>
+        /// Append a CUDA EP instance (configured based on given provider options) to the native OrtSessionOptions instance
+        /// </summary>
+        /// <param name="options">Native OrtSessionOptions instance</param>
+        /// <param name="cudaProviderOptions">Native OrtCUDAProviderOptions instance</param>
+        public delegate IntPtr /*(OrtStatus*)*/DSessionOptionsAppendExecutionProvider_CUDA(
+                                               IntPtr /*(OrtSessionOptions*)*/ options,
+                                               IntPtr /*(const OrtCUDAProviderOptions*)*/ cudaProviderOptions);
+        public static DSessionOptionsAppendExecutionProvider_CUDA SessionOptionsAppendExecutionProvider_CUDA;
```

Comment:
>SessionOptionsAppendExecutionProvider_CUDA [](start = 66, length = 42)

this function's name looks different from the other append EP functions. can it be more consistent?

Reply:
That is because they are fundamentally different. `OrtSessionOptionsAppendExecutionProvider_CUDA` is a symbol available in the shared library if the CUDA EP is built. `SessionOptionsAppendExecutionProvider_CUDA` is available via the C API struct always (whether built with CUDA support or not). 

Also NativeMethods is an "internal concept". If you take a look at the public interface in SessionOptions.cs - the method made available to the user is consistent. There are two overloads of `AppendExecutionProvider_CUDA` - one calls the `OrtSessionOptionsAppendExecutionProvider_CUDA` and the other calls `SessionOptionsAppendExecutionProvider_CUDA` and all these details are abstracted from the user.